### PR TITLE
Translate extmap-allow-mixed between SDP and Jingle.

### PR DESCRIPTION
**Pull request by JonathanLennox**

### CODE REVIEW

**Comment by fippo** [comment]
can you make a spec PR too?

File: modules/sdp/SDP.js [hunk 1]
```diff
@@ -353,6 +353,12 @@ SDP.prototype.toJingle = function(elem, thecreator) {
                 // TODO: handle params
                 elem.up();
             }
+
+            if (SDPUtil.findLine(this.media[i], 'a=extmap-allow-mixed', this.session)) {
+                elem.c('extmap-allow-mixed', {
+                    xmlns: 'urn:xmpp:jingle:apps:rtp:rtp-hdrext:0'
```
Comment (saghul) [inline modules/sdp/SDP.js:359]: Is this ns correct?

**Review by saghul — APPROVED** [review]
LGTM with a question. Not sure what's up with the tests, I'll retry them.

**Comment by saghul** [comment]
Jenkins please test this please.

**Comment by saghul** [comment]
> can you make a spec PR too?

I had a chat with @ralphm and volunteered to participate in a Jingle review effort that will happen. Probably good to keep this in mind.

**Comment by saghul** [comment]
Isn't this related to https://github.com/jitsi/lib-jitsi-meet/pull/2211 ?

File: modules/sdp/SDP.js [hunk 2]
```diff
@@ -353,6 +353,12 @@ SDP.prototype.toJingle = function(elem, thecreator) {
...
+                });
```
Comment (bgrozev) [inline modules/sdp/SDP.js:360]: Do you need an `up()` here?

**Comment by JonathanLennox** [comment]
> can you make a spec PR too?

The PR includes the spec change (though since there's apparently bugs it maybe doesn't cover all the cases?)

File: modules/sdp/SDP.js (hunk 2 above)
Comment (JonathanLennox) [reply modules/sdp/SDP.js:360]: Yes.  (I don't think this is sufficient to fix the bugs tho?)

**Comment by JonathanLennox** [comment]
> can you make a spec PR too?

Do you mean to the XEP?  The latest version of XEP-0294 already includes extmap-allow-mixed.  (Maybe you were thinking of my separate PR for cryptex?)

**Comment by fippo** [comment]
> > can you make a spec PR too?
> 
> Do you mean to the XEP? The latest version of XEP-0294 already includes extmap-allow-mixed. (Maybe you were thinking of my separate PR for cryptex?)

Oh I like what that person from https://xmpp.org/extensions/xep-0294.html#revision-history-v1.1 did but totally forgot :-)

File: modules/sdp/SDP.js [hunk 3]
```diff
@@ -353,6 +353,13 @@ SDP.prototype.toJingle = function(elem, thecreator) {
                 // TODO: handle params
                 elem.up();
             }
+
+            if (SDPUtil.findLine(this.media[i], 'a=extmap-allow-mixed', this.session)) {
```
Comment (nils-ohlmeier) [inline modules/sdp/SDP.js:357]: Nice find with the session level.

**Review by nils-ohlmeier — APPROVED** [review]
LGTM

**Pull request merged**